### ISSUE TITLE
Store websocket events in Dynamo

### ISSUE THREAD

**Issue by dan-muller** [dan-muller/room-app#3]
https://spin.atomicobject.com/2021/01/06/websockets-aws-dynamodb-updates/

**Comment by douglasnaphas** [comment]
I think we'll be better off having just one table for both the WebSocket connection data and the rest of the application data. We might, for example, want to have a global secondary index pointing from the room code back to the connection IDs or active connection IDs, so that we can send a message to every player, like "The game is proceeding!"

Also as you lump more different types of data into one table, it helps spread accesses across the partitions.

**Comment by douglasnaphas** [comment]
The sample code linked in the issue description deletes the connection item from Dynamo on disconnect. I bet we could find a way to use immutable data principles and record journal-style connection and disconnection events.

The most important thing, as always with Dynamo table design, is to list out our desired access patterns.

**Comment by dan-muller** [comment]
I agree with all of your points, I think that I would like to get things working with an assumed "global room" scenario and then separate things off from there.

**Comment by douglasnaphas** [comment]
For getting towards a state where there's a global room, and the backend sends a message to everyone, we probably want to go with a [sparse global secondary index](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/bp-indexes-general-sparse-indexes.html). The partition key on the index could be something like "IsActive," with blank meaning the client for that connection ID has disconnected.

With a sparse GSI, we could avoid [scans](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html).

Basically we want to answer the question, "What are the connection IDs of all the active connections?" by querying `IsActive == 1` instead of scanning the table.

**Comment by dan-muller** [comment]
That makes sense, for the sake of simplicity I was following this guy's lead to get things working but I see your point. 

so we can definitely use a different method to get connections instead of 
```
connections = await db.scan({ TableName: process.env.TABLE_NAME, ProjectionExpression: 'Id' }).promise();
``` 

But do you think the rest of the notification example is relevant?

specifically this part

```
const api = new ApiGatewayManagementApi({
     endpoint: process.env.ENDPOINT,
});
```

What would you say the value of `endpoint` is?
